pkg/provider/kubernetes/gateway/client.go

```diff
@@ -46,7 +46,7 @@ func (reh *resourceEventHandler) OnDelete(obj interface{}) {
 // WatchAll starts the watch of the Provider resources and updates the stores.
 // The stores can then be accessed via the Get* functions.
 type Client interface {
-	WatchAll(namespaces []string, stopCh <-chan struct{}) (<-chan interface{}, error)
+	WatchAll(namespaces []string, experimentalChannel bool, stopCh <-chan struct{}) (<-chan interface{}, error)
```

Comment:
I think this can be moved the structure (and constructor)

Reply:
@ldez I addressed your comment in the last review commit.